### DISCUSSION TITLE
How to change log level?

### DISCUSSION THREAD

**Discussion by rainyl** [Flow-Launcher/Flow.Launcher#2313] [Q&A] [answered]
I am one of the Flow plugin developers, I noticed that the default log level of Flow is INFO, I am wondering how to change the log level to DEBUG in Flow, thus I can debug my plugin more efficient using API.LogDebug() and needn't to worry bothering users because their default log level is INFO.

For now, I can't find some settings to change log level, also, if it is currently not supported, I think it will be nice if added.

Thanks.

**Comment by rainyl** [comment]
Well, after exploring the source code, I find it the DEBUG log level will be used only in debug mode currently, so if I want to see the debug output, I have to build another debug version of Flow, it's really inconvenient, so do you have any plan to support change log level in settings?

**Reply by jjw24** [reply]
Hi there, how's it going?

The log level currently is not a good design, and has not been used properly. 

Generally logging is done only when there is an error, and info level just outputs query running times not code related. 

Debug level logging is not really used at all within flow itself so even with a debug build you wont get much info than what you currently get in the logs.

There is pending work to redo logging completely and when we do we should support the ability to change logging from flow settings UI and expose the level through the API.

**Reply by jjw24** [reply]
In the mean time, feel free to post an issue if you need help figuring out part of flow that's not working for your plugin.

**Reply by rainyl** [reply]
Well, thanks, looks like a long story to write...

I just find it very inconvenient to debug when developing plugins, if I use LogInfo(), much unnecessary information used only when debugging will be written to log file, which will be annoying. Supporting change log level will be useful.

Anyway, thanks for your answers.

**Reply by claudiospizzi** [reply]
I use a small "hack" to provide myself a debug log experience during development:
- Use my own LogDebug method which only write something to the API log method, if I have a debug build
- Use the LogInfo method but trim away the `INFO` and add my `DEBUG` note - of course this will break, if the log output structure changes

```csharp
public void LogDebug(string className, string message, [CallerMemberName] string methodName = "")
{
    #if DEBUG
        _context.API.LogInfo("\b\b\b\b\b\b\b\bDEBUG - " + className, message, methodName);
    #endif
}
```

**Comment by Jack251970** [comment, marked as answer]
Now we support change log level in general settings page: #2739.
